pkg/compressor/compressor.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compressor
+
+import (
+	"compress/gzip"
+	"compress/lzw"
+	"compress/zlib"
+	"io"
+
+	"github.com/sirupsen/logrus"
+)
+
+// CompressSnapshots takes uncompressed data as input and compress the data according to Compression Policy
+// and write the compressed data into one end of pipe.
+func CompressSnapshots(data io.ReadCloser, compressionPolicy string) io.ReadCloser {
+	pReader, pWriter := io.Pipe()
+
+	var gWriter io.WriteCloser
+	logger := logrus.New().WithField("actor", "compressor")
+	logger.Infof("start compressing the snapshot using %v Compression Policy", compressionPolicy)
+
+	switch compressionPolicy {
+	case GzipCompressionPolicy:
+		gWriter = gzip.NewWriter(pWriter)
+
+	case LzwCompressionPolicy:
+		gWriter = lzw.NewWriter(pWriter, lzw.LSB, LzwLiteralWidth)
+
+	case ZlibCompressionPolicy:
+		gWriter = zlib.NewWriter(pWriter)
+
+	}
+
+	go func() {
+		defer pWriter.Close()
+		defer gWriter.Close()
+		defer data.Close()
+		n, err := io.Copy(gWriter, data)
+		if err != nil {
+			logger.Infof("compression failed: %v", err)
+			return
+		}
+		logger.Infof("compressed data size: %v KB", n/1024)
+	}()
+
+	return pReader
+}
+
+// DecompressSnapshots take compressed data and compressionPolicy as input and
+// it decompresses the data according to compression Policy and return uncompressed data.
+func DecompressSnapshots(data io.ReadCloser, compressionPolicy string) io.ReadCloser {
+	var unCompressData io.ReadCloser
+	var err error
+
+	logger := logrus.New().WithField("actor", "de-compressor")
+	logger.Infof("start decompressing the snapshot with %v compressionPolicy", compressionPolicy)
+
+	switch compressionPolicy {
```

Comment:
I think there is no need to add default case because if compressionPolicy is not supported  then [this](https://github.com/gardener/etcd-backup-restore/pull/293/files#diff-7adcf3334be4957b8c200cd4ad81c23a84d4be71d629640291a51ba3b20a198eR46) already gonna handle  that  unsupported compression policy.

Reply:
Additional safety check never hurts :-)